### ISSUE TITLE
Script > File > Save As doesn't work

### ISSUE THREAD

**Issue by batmanasb** [godotengine/godot#4210]
**Operating system or device:**
Linux Mint 17.3
Using the Godot v2.0.1.stable.official binary from the downloads page

**Issue description** (what happened, and what was expected):
When I select a script in the editor and press File > Save As, select a folder to save it to, type in a file name, then press save, nothing happens. Normally this menu option should save a copy of the file as a file with a different name. 

**Steps to reproduce:**
Go to the Script tab, select a script, press File > Save As, type in a new file name, press Save. A new file with that name does not appear.

**Link to minimal example project** (optional but very welcome):

**Comment by neikeq** [comment]
I cannot reproduce this issue on Ubuntu 15.10 64bits with current master branch.

**Comment by akien-mga** [comment]
I can reproduce it with an embedded script on Mageia 6 64bits; if I try to save a script embedded in a SCN scene with "Save As", it will propose <scenename>.res as filename, which is a bug in itself already. Then if I change it to <scenename>.gd and save, nothing happens, the script is not created.

**Comment by akien-mga** [comment]
I don't even know why we allow embedding scripts in binary files, that's awful to maintain...

**Comment by akien-mga** [comment]
@batmanasb Please confirm if your issue was also when using a built-in script instead of one already saved to a file.

**Comment by batmanasb** [comment]
Sorry for the delay, I was asleep. I'm using the Godot 2.0.1 binary and the script was not built-in. Okay so if I make a built-in script and press Ctrl-S to save it, it saves it with the name Godot(some number). Then when I press Save As on any script including itself and name it something like test.gd, it saves the script I just made with that file name. So even though I pressed Save As on an unrelated script, the new file it makes is the built-it in script but with a different name.
